reporter/collector_reporter.go

```diff
@@ -0,0 +1,559 @@
+	stringMap := make(map[string]uint32)
+	stringMap[""] = 0
+
+	// funcMap is a temporary helper that will build the Function array
+	// in profile and make sure information is deduplicated.
+	funcMap := make(map[funcInfo]uint64)
+	funcMap[funcInfo{name: "", fileName: ""}] = 0
+
+	// attributeMap is a temporary helper that maps attribute values to
+	// their respective indices.
+	// This is to ensure that AttributeTable does not contain duplicates.
+	attributeMap := make(map[string]uint64)
+
+	st := profile.SampleType().AppendEmpty()
+	st.SetType(int64(getStringMapIndex(stringMap, "samples")))
+	st.SetUnit(int64(getStringMapIndex(stringMap, "count")))
+
+	pt := profile.PeriodType()
+	pt.SetType(int64(getStringMapIndex(stringMap, "cpu")))
+	pt.SetUnit(int64(getStringMapIndex(stringMap, "nanoseconds")))
+	profile.SetPeriod(1e9 / int64(r.samplesPerSecond))
+
+	locationIndex := uint64(0)
+
+	// Temporary lookup to reference existing Mappings.
+	fileIDtoMapping := make(map[libpf.FileID]uint64)
+
+	for traceKey, traceInfo := range samples {
+		sample := profile.Sample().AppendEmpty()
+		sample.SetLocationsStartIndex(locationIndex)
+
+		sample.SetStacktraceIdIndex(getStringMapIndex(stringMap,
+			traceKey.hash.Base64()))
+
+		slices.Sort(traceInfo.timestamps)
+		startTS = pcommon.Timestamp(traceInfo.timestamps[0])
+		endTS = pcommon.Timestamp(traceInfo.timestamps[len(traceInfo.timestamps)-1])
+
+		sample.TimestampsUnixNano().FromRaw(traceInfo.timestamps)
+		sample.Value().Append(1)
+
+		// Walk every frame of the trace.
+		for i := range traceInfo.frameTypes {
+			frameAttributes := addPdataProfileAttributes(profile, []attrKeyValue{
+				{key: "profile.frame.type", value: traceInfo.frameTypes[i].String()},
+			}, attributeMap)
+
+			loc := profile.Location().AppendEmpty()
+			loc.SetAddress(uint64(traceInfo.linenos[i]))
+			loc.Attributes().FromRaw(frameAttributes)
+
+			switch frameKind := traceInfo.frameTypes[i]; frameKind {
+			case libpf.NativeFrame:
+				// As native frames are resolved in the backend, we use Mapping to
+				// report these frames.
+
+				var locationMappingIndex uint64
+				if tmpMappingIndex, exists := fileIDtoMapping[traceInfo.files[i]]; exists {
+					locationMappingIndex = tmpMappingIndex
+				} else {
+					idx := uint64(len(fileIDtoMapping))
+					fileIDtoMapping[traceInfo.files[i]] = idx
+					locationMappingIndex = idx
+
+					ei, exists := r.executables.Get(traceInfo.files[i])
+
+					// Next step: Select a proper default value,
+					// if the name of the executable is not known yet.
+					var fileName = "UNKNOWN"
+					if exists {
+						fileName = ei.fileName
+					}
+
+					mappingAttributes := addPdataProfileAttributes(profile, []attrKeyValue{
+						// Once SemConv and its Go package is released with the new
+						// semantic convention for build_id, replace these hard coded
+						// strings.
+						{key: "process.executable.build_id.gnu", value: ei.gnuBuildID},
+						{key: "process.executable.build_id.profiling",
```

Comment:
`process.executable.build_id.profiling` already got deprecated. background https://github.com/open-telemetry/semantic-conventions/pull/1520
```suggestion
						{key: "process.executable.build_id.htlhash",
```